zerver/tests/test_signup.py

```diff
@@ -3284,6 +3285,42 @@ def test_create_realm_ldap_email(self) -> None:
                 "newuser_email@zulip.com", self.ldap_password("newuser_with_email")
             )
 
+    @override_settings(OPEN_REALM_CREATION=True)
+    def test_create_realm_race(self) -> None:
+        email = "romeo@zulip.com"
```

Comment:
The `self.nonreg_user` helper is probably a good choice here; we're doing a slow process towards being able to make these `@example.com`, and that means removing hardcoding of `@zulip.com` where possible.